### DISCUSSION TITLE
DFSPH + Volume mapping: avoiding collisions and particle "squeezing".

### DISCUSSION THREAD

**Discussion by twanverweij** [InteractiveComputerGraphics/SPlisHSPlasH#323] [Q&A]
Dear mr. Bender and team,

First of all a bit of gratitude i'm fairly having a blast with the splishsplash code, very well organized and so easy to modify. 

I have a case where I push some particles into a receptacle but i don't want to push them too hard as this gives the pressure solver a hard time. So, to avoid this, I'm looking to confirm some impressions about volume mapping and boundary detection:

- On volume map resolution: part A has dimension X spanning from min_x to max_x. when I apply resolution to the SDF map, will that chop up into cells with width (max_x - min_x)/resolution? And if so, is the grid spaced from the volumetric center or do the cell nodes align with the bounding box around the geometry at say, (min_x, min_y, min_z)? There is some empty space surrounding the actual walls, I'm looking to make the fit a bit better without going too wild with the cell count.

- For a second part B with half the size, will resolution lead to half the cell width of the volumes?

- is a particle allowed to live anywhere between the volume-map (so not the actual geometry) of two parts, given both are half a particle radius apart, or do I need to leave at least two or three particle widths, i.e. should I be able to squeeze a row of particles through a narrow channel?

Thanks in advance,

Twan

**Comment by janbender** [comment]
1. The domain of the grid is a bit larger than the bounding box and is aligned with the enlarged bounding box. You find the code here: 

https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/blob/08f9fc8fad5efb5e346b8b6be4bfaa42be3845a2/Simulator/SimulatorBase.cpp#L2585

Btw. the domain is printed when starting the simulation.

2. It won't be exactly the half cell size since the bounding box is extended a bit (see 1)

3. The boundary representation uses a smoothing kernel. The boundary is smoothed out. You can put a particle exactly next to the boundary but probably it will be pushed away a bit due to the kernel influence.

**Reply by twanverweij** [reply]
Dear Jan,

Many thanks, from semi-learned trial and error I find that a full particle-diameter offset captures the geometries very accurately without going all out on the SDF map resolution; geometry dimension / particle radius = res gives good / efficient  results.
